src/app/utils/lossless-json.converter.ts

```diff
@@ -1,7 +1,7 @@
 import BigNumber from 'bignumber.js';
 
 export function losslessParse(json: string): any {
-    const numberRegex = /^\d+$/;
+    const numberRegex = /^\d+(\.\d+)?$/;
```

Comment:
This `losslessParse` may work only if the BNs are stringified on the JSONs (since this `revive` function from `JSON.parse` is called only after the number already got parsed as such), while it seems the current PFS implementation encodes the `capacity` and other BNs as JSON `number`. This may only be working now if the numbers on your test were small (i.e. < 2^53).
We had this issue on the LC, and ended up using `json-bigint` lib to parse these JSONs. Check [here](https://github.com/raiden-network/light-client/blob/a5c749560381efbae817cf9f10a7017409091100/raiden-ts/src/utils/data.ts#L54-L56) for our implementation.

Reply:
Notice you need to use a proper decoder/schema validator if you go the way we did, since it'll ensure these values are kept as `string`s, or use it with your `reviver` function after using the lib to ensure the BNs are kept as strings or `bigint` or whatever.. We already had the schemas in place, so went with it, but you can use the reviver as well if you don't care that much with the typesafety there.

Reply:
As we don't use the numbers from the PFS API, I won't fix it now. I have put it into this issue: #578 
Thanks for finding this!